src/crimson/osd/object_context.h

```diff
@@ -35,6 +38,9 @@ class ObjectContext : public Blocker,
   ObjectState obs;
   std::optional<SnapSet> ss;
   bool loaded : 1;
+  using watch_key_t = std::pair<uint64_t, entity_name_t>;
```

Comment:
A Q: the map key includes 'entity_name_t', which has a wasteful structure (8b+64b). Should we consider
using a compressed form of entity_name? is the _type required as part of the key?

Reply:
My view is that the increased memory footprint of map's node doesn't affect `sizeof(ObjectContext)`. It would become visible with huge number of watchers which is something I don't expect.

Reply:
I was thinking about search time. std::map is not "performant" as it is...

Reply:
I think the watch / notify machinery is rather away from hot paths. Atomic ref-counting was used initially.

Reply:
Added a comment.